src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -459,12 +459,10 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
                     // Note: RValueOnly is checked at the beginning of this method. Since we are here we need more than readable.
                     //"this" is readonly in members of readonly structs, unless we are in a constructor.
                     if (!thisref.Type.IsValueType ||
```

Comment:
>thisref.Type.IsValueType [](start = 25, length = 24)

Consider extracting a local. And perhaps remove the `thisref` local since it's only used to calculate `IsValueType`.

Reply:
Done